xarray/tests/test_conventions.py

```diff
@@ -63,7 +63,25 @@ def test_decode_cf_with_conflicting_fill_missing_value() -> None:
         np.arange(10),
         {"units": "foobar", "missing_value": np.nan, "_FillValue": np.nan},
     )
-    actual = conventions.decode_cf_variable("t", var)
+
+    expected_warnings = {
+        (
+            SerializationWarning,
+            "variable 't' has non-conforming 'missing_value' nan defined, "
+            "dropping 'missing_value' entirely.",
+        ),
+        (
+            SerializationWarning,
+            "variable 't' has non-conforming '_FillValue' nan defined, "
+            "dropping '_FillValue' entirely.",
+        ),
+    }
+
+    with pytest.warns(Warning) as winfo:
+        actual = conventions.decode_cf_variable("t", var)
+    actual_warnings = {(warn.category, str(warn.message)) for warn in winfo}
+    assert actual_warnings == expected_warnings
```

Comment:
```suggestion
    with pytest.warns(SerializationWarning, match="non-conforming") as winfo:
        actual = conventions.decode_cf_variable("t", var)
```

Reply:
Yes, this works too. But if I understood the pytest documentation correctly, it might miss the first occasion if we do it like this.

Reply:
In this code two warnings are issued, one with `missing_value` and one with `_FillValue`. The suggested pytest.warns just checks if **at least one** warning is issued. See https://docs.pytest.org/en/7.1.x/how-to/capture-warnings.html#additional-use-cases-of-warnings-in-tests

To really see if both warnings were issued, we'd need to check for two. I'm looking how to shorten this a bit.

Reply:
AH I didn't realize that. Let's go with what you have then!

Reply:
I've compressed it and added a comment, to make it more visible.